src/Uno.UI/UI/Xaml/UIElement.wasm.cs

```diff
@@ -415,7 +421,10 @@ protected virtual void OnVisibilityChanged(Visibility oldValue, Visibility newVi
 			}
 			else
 			{
-				SetStyle("visibility", "hidden");
+				// Note: On wasm when we have an 'hidden' (or 'collapsed') element, its height is used to compute the native 'scrollHeight',
+				// driving the SV to flicker when we scroll while at the bottom of the viewport (if those hidden element would have increase the viewport if visible).
+				// To avoid that, we move the element way out of the visible bounds of the view.
+				SetStyle(("visibility", "hidden"), ("top", "-100000px"), ("left", "-100000px"));
```

Comment:
For performance, we should probably have a specialized typescript method that does that, since it's constant values.